### PR TITLE
Some edits on the REKS recipes

### PR DESCRIPTION
**Pull request by bhourahine**
Some suggested changes and hyperlink adjustments:

* Spell checking and grammar changes
* Possible further explanation of some concepts (please check these are correct)
* Removal of comments about Ehrenfest in REKS section, as electron dynamics and Ehrenfest are mean-field approximations to dynamics, but REKS would provide actual level dynamics (if implemented). I agree with the level hopping parts of the comments.

Please check that I have not changed the sense of the descriptions.

Also, are all of the supplied dftb_in example files used? It looks like most of them are relevant, but not directly referenced in the text.

### CODE REVIEW

File: docs/reks/error-handling.rst [hunk 1]
```diff
@@ -21,11 +22,14 @@ The oscillation of SA-REKS energy in SCC process may appear as follows::
...
-in the SCC process. As a result, oscillating phenomena appear in those cases. To solve this,
-the user can increase the shift value by setting ``Shift`` in ``REKS`` block. Then, it may
-show correct variational SA-REKS energy in the SCC process::
+These oscillations mostly arise from the degeneracy of the frontier orbitals. As
+the difference between frontier orbitals gets smaller, it becomes harder to
+determine the order of the frontier orbitals during the SCC process. As a
+result, oscillating phenomena appear in these cases. To solve this, the user can
+increase the shift value by setting ``Shift`` in ``REKS`` block, which increases
+the separation in energy between the lower and upper orbital, stabilising the
+convergence. Smother variational convergence of the SA-REKS energy can occur in
```
Comment (inseonglee) [inline docs/reks/error-handling.rst]: Maybe Smother seems a typo, smoother?

File: docs/reks/error-handling.rst [hunk 2]
```diff
@@ -41,10 +45,13 @@ show correct variational SA-REKS energy in the SCC process::
    Trip   -8.97307659   1.000000  1.000000  1.00
   --------------------------------------------------
 
-Thus, one can check the FONs become :math:`n_a` ~1.2 and :math:`n_b` ~0.8 in this example. As the
-orbital energies of frontier orbitals are close each other, the user should increase the shift
-value as well as ``MaxSCCIterations``. The oscillation of SA-REKS energy can also occur with
-following irregular pattern::
+Thus, one can check the FONs for the PPS state become :math:`n_a` ~1.2 and
+:math:`n_b` ~0.8 in this example. As the orbital energies of frontier PPS and
+OSS orbitals are close each other, the user should increase the shift value as
```
Comment (inseonglee) [inline docs/reks/error-handling.rst]: I think `frontier PPS and OSS orbitals` are not proper expressions, instead `active orbitals` seems to be a good expression.

File: docs/reks/introduction.rst [hunk 3]
```diff
@@ -61,28 +59,34 @@ following input that follows::
...
-you can only include the PPS state into ``Functional`` block in ``Energy`` block.
+Note that the ``REKS = SSR22`` block is a separate environment in the
+calculation, so is not included in the ``Hamiltonian`` block. In a REKS
+calculation, the ``Hamiltonian`` only controls detailed Hamiltonian settings
+such as use of a range-separated hybrid functional or dispersion
+corrections. Three singlet states can be generated from a (2,2) active
+space. These are the single particle perfectly spin-paired singlet state (PPS),
+an open-shell singlet state (OSS) and the doubly excited singlet state
+(DES). Combining these singlet states, we can find the many-particle singlet
+state which minimises the energy functional. In the single-state REKS case we
```
Comment (inseonglee) [inline docs/reks/introduction.rst]: Three singlet states (PPS, OSS, DES) are many-body state, thus I think `single particle state` is not proper expression.

**Review by inseonglee — COMMENTED** [review]
Thanks for checking the grammar as well as some comments and explanations.

I found One type and two wrong points. PPS and OSS states are many-particle states, not single particle state such as Kohn-Sham orbitals. It seems to be good if this part is modified.

The files provided in _archives/recipes/reks directory are currently examples in used.

basis calculation: single_state_reks, sa_reks, si_sa_reks directories
advanced: nonadiabatic_coupling, relaxed_density, microstate directories
error handling: oscillation_energy, benzene, acetylene directories

**Comment by inseonglee** [comment]
@bhourahine If you agree with my comments, then I will fix these comments after merge.

**Comment by bhourahine** [comment]
@inseonglee hopefully the commit addresses your comments

**Pull request merged**